### DISCUSSION TITLE
How to focus a window?

### DISCUSSION THREAD

**Discussion by vdegenne** [WayfireWM/wayfire#2528] [Q&A] [answered]
When I type `google-chrome ...` in a terminal in a viewport, wayfire will focus Google Chrome automatically if it is in another viewport when the tab opens.
How to focus any given window that way? Do I need to send a special signal?
If possible an easy solution without the use of IPC would be ideal.

Thanks

**Comment by soreau** [comment, marked as answer]
> When I type `google-chrome ...` in a terminal in a viewport, wayfire will focus Google Chrome automatically if it is in another viewport when the tab opens. How to focus any given window that way? Do I need to send a special signal?

Typically the app will self-focus the 'parent' window if it is run multiple times and is programmed this way.

> If possible an easy solution without the use of IPC would be ideal.

If you want your application to behave this way, you must program it as such. If you just want to focus a window arbitrarily, you can either use wf-panel window-list (or the equivalent in another panel like waybar), to click on the window in the panel and it will become focused. Otherwise, the next best thing would be a wayfire IPC script.

**Reply by vdegenne** [reply]
Thanks for your answer. It is not an application I made, I just want to focus Alacritty if it is already opened somewhere.
I almost had it using `wmctrl -a Alacritty` but for some reasons, `wmctrl` can't detect my Alacritty windows.

**Reply by soreau** [reply]
https://git.sr.ht/~brocellous/wlrctl
